control-plane/catalog/to-consul/resource.go

```diff
@@ -256,9 +278,21 @@ func (t *ServiceResource) doDelete(key string) {
 // Run implements the controller.Backgrounder interface.
 func (t *ServiceResource) Run(ch <-chan struct{}) {
 	t.Log.Info("starting runner for endpoints")
+	// Register a controller for Endpoints which subsequently registers a
+	// controller for the Ingress resource.
 	(&controller.Controller{
-		Log:      t.Log.Named("controller/endpoints"),
-		Resource: &serviceEndpointsResource{Service: t, Ctx: t.Ctx},
+		Resource: &serviceEndpointsResource{
+			Service: t,
+			Ctx:     t.Ctx,
+			Log:     t.Log.Named("controller/endpoints"),
+			Resource: &serviceIngressResource{
```

Comment:
The problem here is setting it to nil causes nil pointer exceptions in the logs because of the nature of some of these interfaces. The resource cannot be nil. It should not lead to a performance penalty though.